File: src/main/java/allegro/movieplatform/controller/MovieController.java
package allegro.movieplatform.controller;

import allegro.movieplatform.model.Movie;
import allegro.movieplatform.service.MovieService;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
class MovieController {

    @Autowired
    private MovieService service;

    @GetMapping("/movies")
    @ApiOperation(value = "Find all movies",
            response = Movie.class,
            responseContainer = "List",
            httpMethod = "GET")
     ResponseEntity<List<Movie>> getAllMovies(){
        return service.getAllMovies();
    }

    @PostMapping("/movies")
    @ApiOperation(value = "Add movie",
            httpMethod = "POST")
    ResponseEntity<Void> addMovie(@RequestBody Movie movie){
        return service.addMovie(movie);
    }

    @GetMapping("/movies/{title}")
    @ApiOperation(value = "Find movie by title",
            response = Movie.class,
            httpMethod = "GET")
    ResponseEntity<Movie> getMovie(@PathVariable String title){
        return service.getMovieByTitle(title);
    }
}

File: kotlin/src/main/kotlin/milunas/allegromovies/controller/MovieController.kt
package milunas.allegromovies.controller

import milunas.allegromovies.model.Movie
import milunas.allegromovies.service.MovieService
import org.springframework.http.HttpHeaders
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/movie")
class MovieController(private val movieService: MovieService) {

    @GetMapping
    @ResponseStatus(HttpStatus.ACCEPTED)
    fun getAllMovies() : List<Movie>{
        return movieService.getAllMovies()
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun addMovie(@RequestBody movie: Movie) {
        val addedMovie = movieService.addMovie(movie)
        HttpHeaders().set("Location", addedMovie.id.toString())
    }

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.ACCEPTED)
    fun getMovieById(@PathVariable id: String) : Movie{
        return movieService.getMovieById(id.toLong())
    }

}